From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hans Verkuil <hverkuil-cisco@xs4all.nl>
Date: Mon, 15 Mar 2021 08:13:14 +0100
Subject: [PATCH] v4l2-tpg.patch: rename min/max defines to tpg_min/max

With gcc 11 the min/max defines clash with what gcc provides,
so rename these defines. They are only used in the min3/max3
defines, so the change it kept to a minimum.

Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>
Reported-by: Gregor Jasny <gjasny@googlemail.com>
Upstream-Status: Backport
Reference: https://git.linuxtv.org/v4l-utils.git/commit/?id=02f295d4c4
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 utils/common/v4l2-tpg.h     | 10 ++++------
 utils/common/v4l2-tpg.patch | 12 +++++-------
 2 files changed, 9 insertions(+), 13 deletions(-)

diff --git a/utils/common/v4l2-tpg.h b/utils/common/v4l2-tpg.h
index c07c60a..a86ba02 100644
--- a/utils/common/v4l2-tpg.h
+++ b/utils/common/v4l2-tpg.h
@@ -26,12 +26,10 @@ typedef __s8 s8;
 #define pr_info printf
 #define noinline
 
-#ifndef min
-#define min(a,b)	((a) < (b) ? (a) : (b))
-#define max(a,b)	((a) > (b) ? (a) : (b))
-#endif /* !min */
-#define min3(x, y, z) min((typeof(x))min(x, y), z)
-#define max3(x, y, z) max((typeof(x))max(x, y), z)
+#define tpg_min(a,b)	((a) < (b) ? (a) : (b))
+#define tpg_max(a,b)	((a) > (b) ? (a) : (b))
+#define min3(x, y, z) tpg_min((typeof(x))tpg_min(x, y), z)
+#define max3(x, y, z) tpg_max((typeof(x))tpg_max(x, y), z)
 #define array_size(a, b) ((a) * (b))
 #define array3_size(a, b, c) ((a) * (b) * (c))
 
diff --git a/utils/common/v4l2-tpg.patch b/utils/common/v4l2-tpg.patch
index 3224d93..4e85474 100644
--- a/utils/common/v4l2-tpg.patch
+++ b/utils/common/v4l2-tpg.patch
@@ -1,6 +1,6 @@
 --- utils/common/v4l2-tpg.h.bak	2016-04-22 09:22:11.168936563 +0200
 +++ utils/common/v4l2-tpg.h	2016-04-22 09:23:07.337262639 +0200
-@@ -20,12 +20,56 @@
+@@ -20,12 +20,54 @@
  #ifndef _V4L2_TPG_H_
  #define _V4L2_TPG_H_
  
@@ -27,12 +27,10 @@
 +#define pr_info printf
 +#define noinline
 +
-+#ifndef min
-+#define min(a,b)	((a) < (b) ? (a) : (b))
-+#define max(a,b)	((a) > (b) ? (a) : (b))
-+#endif /* !min */
-+#define min3(x, y, z) min((typeof(x))min(x, y), z)
-+#define max3(x, y, z) max((typeof(x))max(x, y), z)
++#define tpg_min(a,b)	((a) < (b) ? (a) : (b))
++#define tpg_max(a,b)	((a) > (b) ? (a) : (b))
++#define min3(x, y, z) tpg_min((typeof(x))tpg_min(x, y), z)
++#define max3(x, y, z) tpg_max((typeof(x))tpg_max(x, y), z)
 +#define array_size(a, b) ((a) * (b))
 +#define array3_size(a, b, c) ((a) * (b) * (c))
 +
